tools/pubsys/src/aws/ami/register/mk_amispec.rs

```diff
@@ -0,0 +1,833 @@
+            .virtualization_type(VirtualizationType::Hvm)
+            .block_device_mapping(BlockDeviceMapping::builder()
+                .ebs(
+                    ebs::Gp2::builder()
+                        .snapshot_id(ROOT_SNAP_ID)
+                        .volume_size(100).unwrap()
+                        .delete_on_termination(true)
+                        .build()
+                )
+                .build()
+                .with_device_name("/dev/xvda"))
+            .build();
+        "default amispec template, no data snapshot"
+    )]
+    #[test_case(
+        AmiArgs {
+            os_image: PathBuf::new(),
+            data_image: None,
+            variant_manifest: VariantManifest::new(
+                toml::toml! {
+                    [package]
+                    name = "test-variant"
+
+                    [package.metadata.build-variant]
+                }.try_into().unwrap(),
+            ),
+            uefi_data: Default::default(),
+            arch: "x86_64".into(),
+            name: "test-ami".into(),
+            description: Some("test description".into()),
+            no_progress: true,
+            regions: Vec::new(),
+            amispec_file: None,
+            ami_output: None,
+        },
+        SPLIT_LAYOUT_SNAPSHOTS.clone(),
+        AmiSpec::builder()
+            .name("test-ami")
+            .architecture(Architecture::X86_64)
+            .ena_support(true)
+            .description("test description")
+            .root_device_name("/dev/xvda")
+            .sriov_net_support(SriovNetSupport::Simple)
+            .virtualization_type(VirtualizationType::Hvm)
+            .block_device_mapping(BlockDeviceMapping::builder()
+                .ebs(
+                    ebs::Gp2::builder()
+                        .snapshot_id(ROOT_SNAP_ID)
+                        .volume_size(DEFAULT_ROOT_VOLUME_SIZE).unwrap()
+                        .delete_on_termination(true)
+                        .build()
+                )
+                .build()
+                .with_device_name("/dev/xvda"))
+           .block_device_mapping(BlockDeviceMapping::builder()
+                .ebs(
+                    ebs::Gp2::builder()
+                        .snapshot_id(DATA_SNAP_ID)
+                        .volume_size(DEFAULT_DATA_VOLUME_SIZE).unwrap()
+                        .delete_on_termination(true)
+                        .build()
+                )
+                .build()
+                .with_device_name("/dev/xvdb"))
+            .build();
+        "default amispec template, with data snapshot"
+    )]
+    #[test_case(
+        AmiArgs {
+            os_image: PathBuf::new(),
+            data_image: None,
+            variant_manifest: VariantManifest::new(
+                toml::toml! {
+                    [package]
+                    name = "test-variant"
+
+                    [package.metadata.build-variant.image-features]
+                    uefi-secure-boot = true
```

Comment:
A different test case relevant to my interests would be to set `uefi-secure-boot = true`, and pair it with an ami spec template like:
```
tpm-support = "v2.0"
boot-mode = "uefi"
```

To confirm that the default `boot-mode = "uefi-preferred"` gets replaced.